synapse/storage/engines/_base.py

```diff
@@ -32,7 +32,7 @@ class IncorrectDatabaseSetup(RuntimeError):
 
 
 class BaseDatabaseEngine(Generic[ConnectionType], metaclass=abc.ABCMeta):
-    def __init__(self, module, database_config: Mapping[str, Any]):
+    def __init__(self, module):
```

Comment:
I can see that one might want database config that applies independently of your choice of DB (or DB driver, for that matter). So I don't mind dropping this change if people aren't convinced.

Reply:
> I can see that one might want database config that applies independently of your choice of DB (or DB driver, for that matter).

Given your reasoning, I'm in favour of leaving the config parameter in, even if we don't use it currently.